apache/superset
# feat: Shows user charts by default when editing a dashboard

### SUMMARY
https://github.com/apache/superset/pull/20684 introduced a change to allow a user to add charts owned/created by others to their dashboards. While this is a reasonable use case, the way it was implemented was to always display all charts accessible to the user which resulted in problems for organizations that contain many charts. From this [comment](https://github.com/apache/superset/pull/20684#issuecomment-1307873917), we can see some of the user feedback:

> To me the experience is degraded, because almost all charts I add are created by me, and there’s a clear candidate for the next chart to add, i.e., it’s the chart most recently created by me.
> 
> Of course it’s good to be able to add charts created by others, but I would think that maybe this could be visible through a toggle (default is to show charts created by me, with toggle in “on” stage for “filter to my charts”).
> 
> This change of removing the filter might be an improvement in a small company, but in a big place it’s very unlikely that charts by others are useful to me, there’s just too many.

This PR modifies the dashboard editor to show by default only the charts owned or created by the user and gives the user an option to show other charts that are accessible to them. The reason for showing user charts by default is because it's more likely they will interact with their own charts and also for performance reasons, as we don't need to load all the metadata associated with a big number of charts upfront.

This PR also:
- Rewrites `actions/sliceEntities` using Typescript to improve type checking and also simplify its logic
- Removes items loaded in the Redux store after the user leaves editing mode

While developing, it was clear that we should also rewrite `components/SliceAdder` to simplify its logic and get rid of the `react-virtualized-auto-sizer` and `react-search-input` dependencies but I'll leave this to a follow up to simplify the review process.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
https://user-images.githubusercontent.com/70410625/229212005-5d784ff4-7290-49d5-a398-729f6dc406a7.mov

### TESTING INSTRUCTIONS
1 - Edit a dashboard
2 - Make sure it displays by default only charts owned or created by the user
3 - Select the option to view other charts
4 - Make sure it displays all charts accessible to the user

### ADDITIONAL INFORMATION
